testsuite/summarize_tests.py

```diff
@@ -71,10 +71,15 @@ def parse_result_file(fname):
 
     for pfile in sorted(glob.glob(os.path.join(test_outdir, "*.xml"))):
         ph_name = os.path.splitext(os.path.split(pfile)[1])[0].replace("_", " ")
-        ph_res = parse_result_file(pfile)
-        results[ph_name] = ph_res
-        for k, v in ph_res.items():
-            totals[k] += v
+        try:
+            ph_res = parse_result_file(pfile)
+            results[ph_name] = ph_res
+            for k, v in ph_res.items():
+                totals[k] += v
+        except Exception as err:
+            msg = f"ERROR: {pfile} not parsable with error {err}"
+            results[ph_name] = {"Tests": 0, "Skipped": 0, "Failures": 0, "Errors": 0, "Time": 0, "Failed tests": [msg]}
+            totals["Failed tests"].append(msg)
```

Comment:
Why is this exception not handled inside `parse_result_file()`? If the exception can be handled as is done here, it should be handled where it is first encountered; otherwise every caller of `parse_result_file` needs to implement it. Seems the results can be handled quite symmetrically to the "normal" return values, which would be even more reason to move the `try … except` inside.

Reply:
Done with next push.